lib/portlayer/network/context.go

```diff
@@ -391,49 +395,60 @@ func (c *Context) Scopes(idName *string) ([]*Scope, error) {
 	c.Lock()
 	defer c.Unlock()
 
-	return c.findScope(idName)
+	return c.findScopes(idName)
 }
 
 func (c *Context) DefaultScope() *Scope {
 	return c.defaultScope
 }
 
-func (c *Context) BindContainer(h *exec.Handle) error {
+func (c *Context) BindContainer(h *exec.Handle) ([]*Endpoint, error) {
 	c.Lock()
 	defer c.Unlock()
 
-	con, ok := c.containers[h.Container.ID]
-	if !ok {
-		return ResourceNotFoundError{error: fmt.Errorf("container %s not found", h.Container.ID)}
+	var con *Container
+	var err error
+
+	if len(h.ExecConfig.Networks) == 0 {
+		return nil, fmt.Errorf("nothing to bind")
 	}
 
-	scopes := con.Scopes()
-	if len(scopes) == 0 {
-		return nil // container not part of any scopes
+	con, ok := c.containers[h.Container.ID]
+	if ok {
+		return nil, fmt.Errorf("container %s already bound", h.Container.ID)
 	}
 
-	var err error
-	var bound []*Scope
-	defer func() {
-		if err == nil {
-			return
+	con = &Container{id: h.Container.ID}
+	for _, ne := range h.ExecConfig.Networks {
+		var s *Scope
+		s, ok := c.scopes[ne.Network.Name]
+		if !ok {
+			return nil, &ResourceNotFoundError{}
 		}
 
-		for _, s := range bound {
-			s.unbindContainer(con)
-		}
-	}()
+		defer func() {
```

Comment:
To roll back the `addContainer` calls (below) that have happened to date in case of error.